lightning/src/ln/channelmanager.rs

```diff
@@ -728,6 +729,9 @@ pub struct OptionalOfferPaymentParams {
 	/// will ultimately fail once all pending paths have failed (generating an
 	/// [`Event::PaymentFailed`]).
 	pub retry_strategy: Retry,
+	/// Contact secrets to include in the invoice request for BLIP-42 contact management.
+	/// If provided, these secrets will be used to establish a contact relationship with the recipient.
```

Comment:
This needs to be substantially more filled-out, including information about intended UX and UI integration logic, how/when to derive secrets, etc.